src/controllers/dex/tokens/index.ts

```diff
@@ -1,42 +1,112 @@
+import { and, eq, sql, isNotNull } from 'drizzle-orm';
+import { alias } from 'drizzle-orm/pg-core';
 import { Router, Request, Response } from 'express';
+import { bignumber } from 'mathjs';
 
 import { DEFAULT_CACHE_TTL } from 'config/constants';
+import { db } from 'database/client';
+import { chains } from 'database/schema/chains';
+import { tokens } from 'database/schema/tokens';
+import { getLastPrices } from 'loader/price';
+import { validateChainId } from 'middleware/network-middleware';
 import { maybeCacheResponse } from 'utils/cache';
-import { toResponse } from 'utils/http-response';
+import { toPaginatedResponse, toResponse } from 'utils/http-response';
+import { prettyNumber } from 'utils/numbers';
+import { createApiQuery, OrderBy, validatePaginatedRequest } from 'utils/pagination';
 import { asyncRoute } from 'utils/route-wrapper';
 
 const router = Router();
 
-// indexer.sovryn.app/dex/30/tokens
+// Endpoint to fetch tradeable tokens with price
 router.get(
   '/',
-  asyncRoute(async (req: Request, res: Response) =>
-    maybeCacheResponse(
+  asyncRoute(async (req: Request, res: Response) => {
+    const chainId = validateChainId(req, true);
+    const p = validatePaginatedRequest(req);
+
+    return maybeCacheResponse(
       res,
-      `/dex/${req.network.chainId}/tokens`,
+      `/dex/${chainId}/tokens`,
       async () => {
-        // todo: load swapable tokens from database...
-        return [];
+        const quoteToken = alias(tokens, 'quote_token');
+        const chain = alias(chains, 'chain');
+
+        const tokenQuery = db
+          .select({
+            id: tokens.id,
+            symbol: tokens.symbol,
+            name: tokens.name,
+            decimals: tokens.decimals,
+            chainId: tokens.chainId,
+            address: tokens.address,
+            logoUrl: tokens.logoUrl,
+            stablecoinId: sql<number>`${quoteToken.id}`.as('stablecoinId'),
+          })
+          .from(tokens)
+          .where(
+            and(
+              chainId ? eq(tokens.chainId, chainId) : undefined,
```

Comment:
required filter by chain id
```suggestion
             eq(tokens.chainId, chainId),
```